Neos.ContentRepository.Core/Classes/Projection/ContentGraph/NodePath.php

```diff
@@ -90,19 +116,27 @@ public function getDepth(): int
         if (!$this->isAbsolute()) {
             throw new \RuntimeException(sprintf(
                 'Depth of relative node path "%s" cannot be determined',
-                $this->value
+                $this->path
             ), 1548162166);
         }
         return count($this->getParts());
     }
 
     public function equals(NodePath $other): bool
     {
-        return $this->value === $other->value;
+        return $this->path === $other->path
+            && $this->rootNodeTypeName?->value === $other->rootNodeTypeName?->value;
+    }
+
+    public function __toString(): string
+    {
+        return $this->rootNodeTypeName
+            ? rtrim('/<' . $this->rootNodeTypeName->value . '>/' . (ltrim($this->path, '/')), '/')
+            : $this->path;
     }
```

Comment:
i thought we wanted to ban `__toString` from value objects? see @bwaidelich 's #4156

Reply:
you're right, what I wanted to do was toString(), not __toString() /facepalm
(alternatively something like serializeToString() or stringSerialize()

Reply:
has been adjusted, we still need string serialization for absolute node paths, though